### PR TITLE
Evaluations reporters

### PR DESCRIPTION
**Pull request by hannako**

### CODE REVIEW

**Review by leenagupte — COMMENTED** [review]
This looks good. A nice bit of figuring out of the Google docs to get the buckets working, especially with the odd Google time format.

The bit I tripped over was the the last commit and the use of `partition_date` My thoughts spiraled in the comments a bit, but they came down to, can we just make `partition_date` public in SampleQuerySet as they need to be accessed outside of the class, and can we just adjust the filename to add the `01` to the partition date in the file name? We could then leave much of the rest of the code unchanged.

File: app/services/discovery_engine/quality/gcp_bucket_exporter.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module DiscoveryEngine::Quality
+  class GcpBucketExporter
+    PROJECT_NAME = Rails.application.config.google_cloud_project_id
```
[the remark this answers is not in the document]
Comment (hannako) [reply app/services/discovery_engine/quality/gcp_bucket_exporter.rb:3]: Oh have I used the wrong variable 🤔 let me check!

[the remark this answers is not in the document]
Comment (hannako) [reply app/services/discovery_engine/quality/gcp_bucket_exporter.rb:3]: No I think this is OK, on integration (and similarly in the higher environments):
```
env | grep "GOOGLE_CLOUD_PROJECT_ID"
GOOGLE_CLOUD_PROJECT_ID=search-api-v2-integration
```
and this is pulled in by [Rails.application.config.google_cloud_project_id](https://github.com/alphagov/search-api-v2/blob/7389bf08e953678d516327cb0bf2c8c9f0e55711/config/application.rb#L22)

[the remark this answers is not in the document]
Comment (hannako) [reply app/services/discovery_engine/quality/gcp_bucket_exporter.rb:3]: I think it's explained [here](https://github.com/alphagov/govuk-docker/pull/815/commits/8b4f7ecf4b3db3fb7f49013fa6d387460218dbf3)

[the remark this answers is not in the document]
Comment (emmalowe) [reply app/services/discovery_engine/quality/gcp_bucket_exporter.rb:3]: Hmm.. I don't understand that. It says we're doing it for consistency, but it seems to me like we've made it inconsistent? What happens if we try to run this rake task locally? Won't it go to a different bucket than if we run it in integration?

**Review by leenagupte — COMMENTED** [review]
I should have asked in the previous review, but to double check my understanding of these changes. In the new world, the rake tasks to run the evaluations (excluding create the sample query sets) will only call the EvaluationRunner class? The EvaluationRunner class will do all of the work to set up the Prometheus Client and metrics collector?

If that's the case, then in think the general structure of the changes looks good. Those partition dates are still a bit of head scratcher, but that can be revisited at a later date.

I've added some comments, most aren't blockers.

I haven't approved as Emma made a good suggestion about adding tests for the date edge cases.

It would also be great if the commits could be split up a bit. There's a lot of scrolling needed to check a change against the corresponding tests.

File: app/services/discovery_engine/quality/partition_date.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    def self.calculate
+      new(month_label:, month:, year:).calculate
+    end
+
+    def calculate
+      Date.new(
+        date_components.year,
+        date_components.month,
+        date_components.day,
+      )
```
[the remark this answers is not in the document]
Comment (hannako) [reply app/services/discovery_engine/quality/partition_date.rb:18]: Do you! I find this so much easier to follow, and I like that PartitionDate returns a date object which is much more useful.
